surface/include/pcl/surface/impl/marching_cubes.hpp

```diff
@@ -246,7 +246,7 @@ pcl::MarchingCubes<PointNT>::performReconstruction (pcl::PointCloud<PointNT> &po
   voxelizeData ();
 
   // preallocate memory assuming a hull. suppose 6 point per voxel
-  double size_reserve = std::min((double) intermediate_cloud.points.max_size (),
+  double size_reserve = std::min((double) intermediate_cloud.max_size (),
```

Comment:
This is the line causing error. I've seen a similar fix (by @mvieth I think) regarding `std::min` or `std::max` where using them introduced an error.

The testing was done here:
https://kunaltyagi.visualstudio.com/pcl/_build?definitionId=3&_a=summary&repositoryFilter=2&branchFilter=99%2C99%2C99%2C99%2C99%2C99

Reply:
Not sure what fix you are referring to, but something like this seems better: `const std::size_t size_reserve = std::min(intermediate_cloud.max_size (), 2 * 6 * (std::size_t) (res_y_*res_z_ + res_x_*res_z_ + res_x_*res_y_));`

Reply:
Isolate this change to another PR so this once can be merged.